html/includes/api_functions.inc.php

```diff
@@ -244,6 +245,10 @@ function get_graph_generic_by_hostname()
     rrdtool_initialize(false);
     include 'includes/graphs/graph.inc.php';
     rrdtool_close();
+
+    if ($vars['output'] === 'base64') {
+        api_success(['image' => $base64_output, 'content-type' => get_image_type()], 'image');
```

Comment:
Do you need to add this to the other functions which return graphs?

Reply:
Yep, does the format and names seem ok? If so, i'll append it to the port methods as well